sys-devel/llvm/patches/libunwind-12.0.1.patchset

```diff
@@ -0,0 +1,416 @@
+   bool             _unwindInfoMissing;
+   bool             _isSignalFrame;
+-#if defined(_LIBUNWIND_TARGET_LINUX) && defined(_LIBUNWIND_TARGET_AARCH64) || defined(__HAIKU__)
++#if defined(_LIBUNWIND_TARGET_LINUX) && defined(_LIBUNWIND_TARGET_AARCH64) \
++    || defined(_LIBUNWIND_TARGET_HAIKU) && defined(_LIBUNWIND_TARGET_X86_64)
+   bool             _isSigReturn = false;
+ #endif
+ };
+@@ -1898,7 +1901,8 @@ bool UnwindCursor<A, R>::getInfoFromSEH(pint_t pc) {
+ 
+ template <typename A, typename R>
+ void UnwindCursor<A, R>::setInfoBasedOnIPRegister(bool isReturnAddress) {
+-#if defined(_LIBUNWIND_TARGET_LINUX) && defined(_LIBUNWIND_TARGET_AARCH64) || defined(__HAIKU__)
++#if defined(_LIBUNWIND_TARGET_LINUX) && defined(_LIBUNWIND_TARGET_AARCH64) \
++    || defined(_LIBUNWIND_TARGET_HAIKU) && defined(_LIBUNWIND_TARGET_X86_64)
+   _isSigReturn = false;
+ #endif
+ 
+@@ -2000,7 +2004,8 @@ void UnwindCursor<A, R>::setInfoBasedOnIPRegister(bool isReturnAddress) {
+   }
+ #endif // #if defined(_LIBUNWIND_SUPPORT_DWARF_UNWIND)
+ 
+-#if defined(_LIBUNWIND_TARGET_LINUX) && defined(_LIBUNWIND_TARGET_AARCH64) || defined(__HAIKU__)
++#if defined(_LIBUNWIND_TARGET_LINUX) && defined(_LIBUNWIND_TARGET_AARCH64) \
++    || defined(_LIBUNWIND_TARGET_HAIKU) && defined(_LIBUNWIND_TARGET_X86_64)
+   if (setInfoForSigReturn())
+     return;
+ #endif
+@@ -2069,7 +2074,7 @@ int UnwindCursor<A, R>::stepThroughSigReturn(Registers_arm64 &) {
+   _isSignalFrame = true;
+   return UNW_STEP_SUCCESS;
+ }
+-#elif defined(__HAIKU__)
++#elif defined(_LIBUNWIND_TARGET_HAIKU) && defined(_LIBUNWIND_TARGET_X86_64)
+ 
+ #include <commpage_defs.h>
+ #include <signal.h>
+@@ -2082,11 +2087,11 @@ template <typename A, typename R>
+ bool UnwindCursor<A, R>::setInfoForSigReturn() {
+ #if defined(_LIBUNWIND_TARGET_X86_64)
+   addr_t signal_handler = (((addr_t*)__gCommPageAddress)[COMMPAGE_ENTRY_X86_SIGNAL_HANDLER] + (addr_t)__gCommPageAddress);
+-	addr_t signal_handler_ret = signal_handler + 45;
++  addr_t signal_handler_ret = signal_handler + 45;
+ #endif
+   pint_t pc = static_cast<pint_t>(this->getReg(UNW_REG_IP));
+   if (pc == signal_handler_ret) {
+-  	//printf("signal frame detected\n");
++    //printf("signal frame detected\n");
+     _info = {};
+     _info.start_ip = signal_handler;
+     _info.end_ip = signal_handler_ret;
+@@ -2098,16 +2103,16 @@ bool UnwindCursor<A, R>::setInfoForSigReturn() {
+ 
+ template <typename A, typename R>
+ int UnwindCursor<A, R>::stepThroughSigReturn() {
+-	//printf("stepThroughSigReturn\n");
++  //printf("stepThroughSigReturn\n");
+   _isSignalFrame = true;
+   pint_t sp = _registers.getSP();
+- // printf("sp: %p\n", (void*)sp);
++  // printf("sp: %p\n", (void*)sp);
+ #if defined(_LIBUNWIND_TARGET_X86_64)
+   vregs *regs = (vregs*)(sp + 0x70);
+   //printf("&regs: %p\n", regs);
+- 
++
+   _registers.setRegister(UNW_REG_IP, regs->rip);
+-  _registers.setRegister(UNW_REG_SP, regs->rsp);  
++  _registers.setRegister(UNW_REG_SP, regs->rsp);
```

Comment:
Can you avoid unnecessary whitespace differences please? (There's a way to get git to `add` only non-whitespace changes, I think. Or you can use `git gui` to do per-hunk staging.)

Reply:
Done.